app/services/measure_unit_service.rb

```diff
@@ -9,9 +9,15 @@ def call
       unit_qualifier_code = unit[:measurement_unit_qualifier_code]
       unit_key = "#{unit_code}#{unit_qualifier_code}"
 
-      if acc[unit_key].blank?
-        acc[unit_key] = {}
-        acc[unit_key] = MeasurementUnit.measurement_unit(unit_code, unit_key)
+      next if acc[unit_key].present?
```

Comment:
does this serve to filter out the acc having unit_key?
can we do it before the loop?
`units.reject{|k,v| v[:measurement_unit_code] || v[:measurement_unit_qualifier_code] }`
so we can simplify it?

Reply:
No it does not filter. It avoids having to accumulate already accumulated measure units on the same measure.